react/src/Root/App/DistrictOverview/index.tsx

```diff
@@ -49,21 +72,102 @@ const requests: { [key: string]: ClientAttributes<Props, Params> } = {
 
-function wrapInArray<T>(item?: T) {
-    if (isNotDefined(item)) {
-        return [];
-    }
+const getNewMapStateOnRiskPointHoverChange = (
+    mapState: MapStateElement[],
+    catPointList: RiskPoint[] = [],
+    id: number | undefined,
+    featureIdentifier: FeatureIdentifiers,
+    featureFromIdentifier: FeatureFromIdentifier,
+    lineStringIdentifier,
+) => {
+    const newMapState = [] as MapStateElement[];
+    const geosite = id ? featureFromIdentifier[id] : undefined;
 
-    return [item];
-}
+    const setAsLight: {
+        [key: number]: boolean;
+    } = {};
+
+    catPointList.forEach((catPoint) => {
+        const shouldDim = id ? catPoint.geosite !== geosite : false;
+        const rs = catPoint.relocationSites;
+
+        let relocationSites = [] as RelocationSite[];
+        if (rs) {
+            relocationSites = rs;
+        }
+
+        if (!shouldDim) {
+            newMapState.push({
+                id: featureIdentifier[catPoint.geosite],
+                value: { dim: shouldDim },
+            });
+
+            setAsLight[featureIdentifier[catPoint.geosite]] = true;
+        }
+
+        if (shouldDim && !setAsLight[featureIdentifier[catPoint.geosite]]) {
+            newMapState.push({
+                id: featureIdentifier[catPoint.geosite],
+                value: { dim: shouldDim },
+            });
+        }
+
+        relocationSites
+            .filter(r => featureIdentifier[r.code])
+            .forEach((r) => {
+                if (!shouldDim) {
+                    newMapState.push({
+                        id: featureIdentifier[r.code],
+                        value: { dim: shouldDim },
+                    });
+                    newMapState.push({
+                        id: lineStringIdentifier[`${catPoint.geosite}:${r.code}`],
+                        value: { dim: shouldDim },
+                    });
+
+                    setAsLight[featureIdentifier[r.code]] = true;
+                    setAsLight[lineStringIdentifier[`${catPoint.geosite}:${r.code}`]] = true;
+                }
+
+                if (shouldDim && !setAsLight[featureIdentifier[r.code]]) {
+                    newMapState.push({
+                        id: featureIdentifier[r.code],
+                        value: { dim: shouldDim },
+                    });
+                }
+
+                if (shouldDim && !setAsLight[lineStringIdentifier[`${catPoint.geosite}:${r.code}`]]) {
+                    newMapState.push({
+                        id: lineStringIdentifier[`${catPoint.geosite}:${r.code}`],
+                        value: { dim: shouldDim },
+                    });
+                }
+            });
+    });
+
+    return newMapState;
+};
 
 class DistrictOverview extends React.PureComponent<MyProps, State> {
     public constructor(props: MyProps) {
         super(props);
 
-        this.state = {};
+        this.state = {
+            mapState: [],
+        };
     }
 
+    private featureIdentifier = {}
```

Comment:
Shouldn't there be types for `featureIdentifier`, `featureFromIdentifier` and `lineStringIdentifier`